packages/blocks/table/src/Table.tsx

```diff
@@ -61,24 +69,78 @@ const useTableData = (data: AppProps["data"]) => {
   return [tableData, setTableData] as const;
 };
 
-export const App: BlockComponent<AppProps> = ({
-  data = defaultData,
+const getLinkedAggregation = (params: {
+  linkedAggregations: BlockProtocolLinkedAggregation[];
+  path: string;
+  sourceEntityId: string;
+}): BlockProtocolLinkedAggregation | undefined => {
+  const { linkedAggregations, path, sourceEntityId } = params;
+
+  return linkedAggregations.find(
+    (aggregation) =>
+      aggregation.path === path &&
+      aggregation.sourceEntityId === sourceEntityId,
+  );
+};
+
+const cleanUpdateLinkedAggregationAction = (
+  action: BlockProtocolUpdateLinksAction & {
+    updatedOperation: Partial<BlockProtocolLinkedAggregation>;
+  },
+) => {
+  const nextAction = { ...action };
+
+  nextAction.updatedOperation.multiSort =
+    action.updatedOperation.multiSort?.map((sort) => {
+      const newSort = { ...sort };
+      newSort.__typename = undefined;
+      return newSort;
+    });
+
+  nextAction.updatedOperation.pageCount = undefined;
+
+  nextAction.updatedOperation.__typename = undefined;
+
+  return nextAction;
+};
+
+const path = "$.data";
+
+export const Table: BlockComponent<AppProps> = ({
+  aggregateEntities,
+  aggregateEntityTypes,
+  entityId,
   initialState,
+  linkedAggregations,
   schemas,
   updateEntities,
-  entityId,
-  aggregateEntities,
-  aggregateEntityTypes,
+  updateLinks,
 }) => {
-  const [tableData, setTableData] = useTableData(data);
+  const [tableData, setTableData] = useTableData(defaultData);
+
+  const matchingLinkedAggregation = useMemo(() => {
+    return getLinkedAggregation({
+      linkedAggregations: linkedAggregations ?? [],
+      path,
+      sourceEntityId: entityId,
+    });
+  }, [entityId, linkedAggregations]);
+
+  useEffect(() => {
+    setTableData({
+      linkedAggregation: matchingLinkedAggregation,
+      data: matchingLinkedAggregation?.results,
+    });
+  }, [matchingLinkedAggregation, setTableData]);
```

Comment:
Setting state derived from props in an effect like is an anti-pattern. It results in more re-renders than necessary or than you might expect, and can also result in weird renders with different parts of state representing different versions of a component's props, which could cause straight up crashes. 

What you ought to do is save the version of the props you used to generate the state, and just call set state in the render body when these don't match. This would then allow you to remove `defaultData` too, which isn't really needed. I think this probably ought to be rolled into `useTableData` which already has a facility for this, but you're breaking it because `defaultData` is constant. 

A new implementation of `useTableData` that looks like this ought to work (I have not tested this)

```ts
const useTableData = (
  linkedAggregation: BlockProtocolLinkedAggregation | undefined,
) => {
  const defaultTableData: TableData = {
    linkedAggregation,
    data: linkedAggregation?.results,
  };

  const defaultAggregateData = {
    tableData: defaultTableData,
    prevLinkedAggregation: linkedAggregation,
  };

  const [{ prevLinkedAggregation, tableData }, setAggregateTableData] =
    useState(defaultAggregateData);

  const setTableData = useCallback(
    (nextData: TableData) =>
      setAggregateTableData((existing) => ({
        ...existing,
        tableData: nextData,
      })),
    [],
  );

  if (linkedAggregation !== prevLinkedAggregation) {
    setAggregateTableData(defaultAggregateData);
  }

  return [tableData, setTableData] as const;
};
```

And then you can update this here to just pass the linked aggregation that you're forming the default state from

```suggestion
  const matchingLinkedAggregation = useMemo(() => {
    return getLinkedAggregation({
      linkedAggregations: linkedAggregations ?? [],
      path,
      sourceEntityId: entityId,
    });
  }, [entityId, linkedAggregations]);

  const [tableData, setTableData] = useTableData(matchingLinkedAggregation);
```

Reply:
This worked. Thanks!